dev-tools/reconfigurator-cli/tests/test_basic.rs

```diff
@@ -25,3 +43,214 @@ fn test_basic() {
+    let mut s = String::new();
+    swriteln!(s, "load {} {}", saved_state2_path, collection.id);
+    swriteln!(s, "blueprint-save {} {}", new_blueprint.id, new_blueprint_path);
+    std::fs::write(&script2_path, &s)
+        .with_context(|| format!("write {}", &script2_path))
+        .unwrap();
+    let exec = Exec::cmd(path_to_cli()).arg(&script2_path);
+    let (exit_status, _, stderr_text) = run_command(exec);
+    assert_exit_code(exit_status, EXIT_SUCCESS, &stderr_text);
+
+    // Load the blueprint we just wrote.
+    let new_blueprint2: Blueprint = read_json(&new_blueprint_path).unwrap();
+    assert_eq!(new_blueprint, new_blueprint2);
+
+    // Import the new blueprint.
+    let nexus_internal_url =
+        format!("http://{}/", cptestctx.internal_client.bind_address);
+    let nexus_client =
+        nexus_client::Client::new(&nexus_internal_url, log.clone());
+    nexus_client
+        .blueprint_import(&new_blueprint)
+        .await
+        .expect("failed to import new blueprint");
+
+    let found_blueprint = nexus_client
+        .blueprint_view(&new_blueprint.id)
+        .await
+        .expect("failed to find imported blueprint in Nexus")
+        .into_inner();
+    assert_eq!(found_blueprint, new_blueprint2);
+
+    // Set the blueprint as the (disabled) target.
+    nexus_client
+        .blueprint_target_set(&nexus_client::types::BlueprintTargetSet {
+            target_id: new_blueprint.id,
+            enabled: false,
+        })
+        .await
+        .context("setting target blueprint")
+        .unwrap();
+
+    // Read that back.
+    let target = nexus_client
+        .blueprint_target_view()
+        .await
+        .context("fetching target blueprint")
+        .unwrap();
+    assert_eq!(target.target_id, new_blueprint.id);
+
+    // Now clean up the temporary directory.
+    for path in [
+        saved_state1_path,
+        saved_state2_path,
+        script1_path,
+        script2_path,
+        new_blueprint_path,
+    ] {
+        std::fs::remove_file(&path)
+            .with_context(|| format!("remove {}", &path))
+            .unwrap();
+    }
+
+    std::fs::remove_dir(&tmpdir_path)
+        .with_context(|| format!("remove {}", &tmpdir_path))
+        .unwrap();
+}
+
+fn read_json<T: for<'a> serde::Deserialize<'a>>(
+    path: &Utf8Path,
+) -> Result<T, anyhow::Error> {
+    let file = std::fs::OpenOptions::new()
+        .read(true)
+        .open(path)
+        .with_context(|| format!("open {:?}", path))?;
+    serde_json::from_reader(file).with_context(|| format!("read {:?}", path))
+}
+
+fn write_json<T: serde::Serialize>(
+    path: &Utf8Path,
+    obj: &T,
+) -> Result<(), anyhow::Error> {
+    let file = std::fs::OpenOptions::new()
+        .write(true)
+        .create(true)
+        .open(path)
+        .with_context(|| format!("open {:?}", path))?;
+    serde_json::to_writer_pretty(&file, obj)
+        .with_context(|| format!("write {:?}", path))?;
+    Ok(())
```

Comment:
```suggestion
    let file = std::fs::OpenOptions::new()
        .write(true)
        .create(true)
        .open(path)
        .with_context(|| format!("open {}", path))?;
    serde_json::to_writer_pretty(&file, obj)
        .with_context(|| format!("write {}", path))?;
    Ok(())
```

As above, need to use buffering here.